test/functional/model-checkin.js

```diff
@@ -0,0 +1,183 @@
+import { Application } from 'spectron';
+import electron from 'electron-prebuilt';
+import { assert } from 'chai';
+import MockServer from '../util/mock-server';
+import models from '../test-cases/models';
+import students from '../test-cases/students';
+
+describe('Checking a model in', function () {
+
+    this.timeout(10000);
+    let app;
+    let mockServer = new MockServer();
+
+    before(() => {
+        app = new Application({
+            path: electron,
+            args: ['index.js', '--port=8080']
+        });
+        return app.start().then(() => {
+            return mockServer.start(8080);
+        });
+    });
+
+    after(() => {
+        if (app && app.isRunning()) {
+            return app.stop().then(() => {
+                return mockServer.stop();
+            });
+        }
+    });
+
+    beforeEach(() => {
+        mockServer.clearExpectations();
+    });
+
+    it('navigates to the student page', () => {
+        mockServer.expect({
+            method: 'get',
+            endpoint: 'student',
+            qs: {
+                id: '123456'
+            },
+            response: {
+                status: 'success',
+                data: students[2]
+            }
+        });
+        mockServer.expect({
+            method: 'get',
+            endpoint: 'model/all',
+            response: {
+                status: 'success',
+                data: {
+                    models
+                }
+            }
+        });
+        return app.client.keys('999999').then(() => {
```

Comment:
Why does this work? The server is expecting a get request with query string 123456 no 99999